frontend/model/database.js

```diff
@@ -9,12 +9,42 @@ const log = localforage.createInstance({
   storeName: 'Contracts'
 })
 
+const isLightweightClient = process.env.LIGHTWEIGHT_CLIENT === 'true'
+
 // make gi.log use localforage for storage
-sbp('sbp/selectors/overwrite', {
-  'gi.db/get': key => log.getItem(key),
-  // TODO: handle QuotaExceededError
-  'gi.db/set': (key, value) => log.setItem(key, value)
-})
+sbp('sbp/selectors/overwrite',
+  isLightweightClient
+    ? {
+        'gi.db/get': key => {
+          const contractId = sbp('gi.db/log/contractIdFromLogHEAD', key)
+          if (!contractId) {
+            return null
+          }
+          const state = sbp('state/vuex/state')
+          if (state.contracts[contractId]?.HEAD) {
+            return state.contracts[contractId].HEAD
+          }
+          return null
+        },
+        'gi.db/set': (key, value) => {
+          return Promise.resolve(value)
+        },
+        'gi.db/delete': () => {
+          return Promise.resolve()
+        }
+      }
+    : {
+        'gi.db/get': key => {
+          return log.getItem(`${key}`)
```

Comment:
Since `key` should already be a string, there is no need for a template literal here; a type annotation should be enough. The same goes for `'gi.db/delete'` below.